File: models/reservation.py
from sqlalchemy import *
from datetime import datetime

from .base import Base, id_col

class Reservation(Base):
    __tablename__ = 'reservations'
    id = id_col()
    name = Column(String())
    email = Column(String())
    time = Column(DateTime())
    party_size = Column(Integer())

    def __init__(self, name='', email='', time=None, party_size=0, **kwargs):
        self.name = name
        self.email = email
        self.time = time
        self.party_size = party_size

File: view/src/models/Reservation.ts
interface ReservationResponse {
    id: string;
    name: string;
    email: string;
    partySize: number;
    time: string;
}

export default class Reservation {
    id: string = "";
    name: string = "";
    email: string = "";
    partySize: number = 1;
    time: Date = new Date(0);

    fromJson(resp: ReservationResponse): Reservation {
        this.id = resp.id;
        this.name = resp.name;
        this.email = resp.email;
        this.partySize = resp.partySize;
        this.time = new Date(resp.time);
        return this
    }

    toJson() {
        return {
            id: this.id,
            name: this.name,
            email: this.email,
            party_size: this.partySize,
            time: this.time.toISOString()
        }
    }
}